src/rust/src/backend/keys.rs

```diff
@@ -50,13 +65,88 @@ fn load_pem_private_key<'p>(
     unsafe_skip_rsa_key_validation: bool,
 ) -> CryptographyResult<pyo3::Bound<'p, pyo3::PyAny>> {
     let _ = backend;
-    let password = password.as_ref().map(CffiBuf::as_bytes);
-    let mut status = utils::PasswordCallbackStatus::Unused;
-    let pkey = openssl::pkey::PKey::private_key_from_pem_callback(
+
+    let p = x509::find_in_pem(
         data.as_bytes(),
-        utils::password_callback(&mut status, password),
-    );
-    let pkey = utils::handle_key_load_result(py, pkey, status, password)?;
+        |p| ["PRIVATE KEY", "ENCRYPTED PRIVATE KEY", "RSA PRIVATE KEY", "EC PRIVATE KEY", "DSA PRIVATE KEY"].contains(&p.tag()),
+        "Valid PEM but no BEGIN/END delimiters for a private key found. Are you sure this is a private key?"
+    )?;
+    let password = password.as_ref().map(|v| v.as_bytes());
+    let mut password_used = false;
+    // TODO: Surely we can avoid this clone?
+    let tag = p.tag().to_string();
+    let data = match p.headers().get("Proc-Type") {
+        Some("4,ENCRYPTED") => {
+            password_used = true;
+            let Some(dek_info) = p.headers().get("DEK-Info") else {
+                todo!()
+            };
+            let Some((cipher_algorithm, iv)) = dek_info.split_once(',') else {
+                todo!()
+            };
```

Comment:
@alex , could You please make this code (81-86) compatible with Rust 1.63! let...else is unstable in 1.63... :-(

Reply:
We've had an MSRV of 1.65 since cryptography 43.0.0 in July of last year, we're not going to revert that absent exceptional circumstances

Reply:
@alex , I still maintain my custom build based on Rust 1.63 w/o GATs, maturin & workspace deps. It works fast on my retro systems (Win'7 and earlier) and very stable, too. :-)
Repacement:
```
            let dek_info = match p.headers().get("DEK-Info") {
                Some(dek_info) => dek_info,
                None => {
                    todo!()
                },
            };
            let (cipher_algorithm, iv) = match dek_info.split_once(',') {
                Some((cipher_algorithm, iv)) => (cipher_algorithm, iv),
                None => {
                    todo!()
                },
            };
```

Reply:
We're not going out of our way to to support things that we don't support.